SwiftSyft/Classes/SyftWebSocket_iOS12.swift

```diff
@@ -0,0 +1,75 @@
+//
+//  SyftWebSocket.swift
+//  SwiftSyft
+//
+//  Created by Sasha Bataieva on 25/01/2020.
+//
+
+import Foundation
+import Starscream
+
+public class SyftWebSocketIOS12: SyftWebSocket, WebSocketDelegate {
+    var socket: WebSocket!
+
+    public func didReceive(event: WebSocketEvent, client: WebSocket) {
+        switch event {
+        case .connected(let headers):
+            connected()
+            print("websocket is connected: \(headers)")
+        case .disconnected(let reason, let code):
+            disconnect()
+            print("websocket is disconnected: \(reason) with code: \(code)")
+        case .text(let string):
+            sendText(text: string)
+            print("Received text: \(string)")
+        case .binary(let data):
+            send(data: data)
+            print("Received data: \(data.count)")
+        case .cancelled:
+            disconnect()
+        case .error(let error):
+            handleError(error)
+        default: // ping, pong, viablityChanged, reconnectSuggested
+            break
+        }
+    }
+
+    func handleError(_ error: Error?) {
+        if let err = error as? WSError {
+            print("websocket encountered an error: \(err.message)")
+        } else if let err = error {
+            print("websocket encountered an error: \(err.localizedDescription)")
+        } else {
+            print("websocket encountered an error")
+        }
+    }
+
+    // MARK: - SocketClientProtocol
+    required public init(url: URL, pingInterval: Double) {
+        super.init(url: url, pingInterval: pingInterval)
```

Comment:
No more need for `super.init` since you're not inheriting from `SyftWebSocket` anymore.
I don't think you need to save the url as properties since you can instantiate starscream socket here.

You also need to do 2 other things here:
1.) Assert that the URL is wss:// , if not assert a runtime error, maybe `preconditionFailure`
2.) Create a timer that periodically sends a ping via starscream socket: 
References: 
- https://github.com/OpenMined/grid.js/blob/master/SOCKETS.md (check keep alive for the message to send section if starscream does not implement pinging)
-  https://swifting.io/blog/2016/03/13/9-how-to-invalidate-nstimer-properly/ for the proper way to create a timer avoiding retain cycles.

Reply:
For wss connection, I think, we need separate issue, it will take time for creating certificate (Starscream has this functionality, but they have not instructions on how to run wss connection)
And for the timer, we will need a separate issue, it looks like Startscream does not has a notification   for a not connected server. I' m going investigate this scenario this weekend